### PR TITLE
[None][feat] Add conversation params to LLM API

### PR DESCRIPTION
**Pull request by jiaganc**
## Summary by CodeRabbit

* **New Features**
  * Added conversation parameters with conversation ID support enabling stable multi-turn conversation routing across the executor, LLM API, and OpenAI-compatible endpoints. Conversation IDs can be provided via request body or standard HTTP headers.

* **Refactoring**
  * Refactored conversation ID handling in server implementations for improved maintainability.

* **Tests**
  * Added unit tests validating conversation parameter validation and header-based ID resolution logic.

## Description

Route conversation identifiers through top-level `conversation_params` instead of `disaggregated_params`.

This adds public LLM, executor, OpenAI protocol, and serve-edge plumbing for `ConversationParams`, resolves supported session headers into top-level request state, and propagates the Python-only object through the PyExecutor request broadcast path. The conflict resolution preserves the newer `upstream/main` `ctx_usage` handling while moving conversation identity out of disaggregated params.

## Test Coverage

- Commit hooks passed, including `isort`, `yapf`, `ruff`, `ruff-format`, merge-conflict checks, API stability reference checks, and DCO checks.
- `git diff --check upstream/main..HEAD`
- `PYTHONPYCACHEPREFIX=/tmp/main-conv-id-pycache python3 -m py_compile ...` for all changed Python files.
- Added and updated unit/API stability coverage for `ConversationParams`, OpenAI request plumbing, router header resolution, and request broadcasting.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tests/integration/defs/disaggregated/test_workers.py [hunk 1]
```diff
@@ -800,9 +801,10 @@ async def test_explicit_conversation_id(self):
                     req = CompletionRequest(
                         model=self.model_name,
                         prompt=prompt,
+                        conversation_params=ConversationParams(
+                            conversation_id=conv_id),
                         disaggregated_params=DisaggregatedParams(
-                            request_type="context_only",
-                            conversation_id=conv_id))
+                            request_type="context_only"))
```
[the remark this answers is not in the document]
Comment (jiaganc) [reply tests/integration/defs/disaggregated/test_workers.py]: Addressed in 814964c076 by extending tests/integration/defs/disaggregated/test_workers.py so test_explicit_conversation_id covers both supported client inputs: X-Correlation-ID headers and JSON body conversation_params, while keeping the sticky-routing and load-balancing assertions. Validation: commit hooks, git diff --check, and py_compile for changed Python files.

**Review by fredricz-20070104 — APPROVED** [review]
Approved

**Review by fredricz-20070104 — APPROVED** [review]
Approved.

File: tensorrt_llm/_torch/pyexecutor/_util.py [hunk 2]
```diff
@@ -483,6 +483,7 @@ def _create_dummy_mm_context_request(
                             torch.zeros(1, 1, dtype=torch.int32)
                         }
                     }
+            request.py_conversation_params = None
```
Comment (Shixiaowei02) [inline tensorrt_llm/_torch/pyexecutor/_util.py:486]: `py_conversation_params` is threaded through the executor and broadcast across ranks but never actually read there. The only consumer is the serve-edge router, which reads it from the request body before the executor. Given that, would it make sense to drop the executor plumbing entirely?

File: tensorrt_llm/serve/openai_protocol.py [hunk 3]
```diff
@@ -155,14 +156,32 @@ class DisaggregatedParams(OpenAIBaseModel):
...
-    conversation_id: Optional[str] = None
     ctx_usage: Optional[UsageInfo] = None
     # TODO(TRTLLM-12407): Multimodal E/PD over trtllm-serve needs these protocol fields too:
     # encoder embedding handles, multimodal hashes, and optional mRoPE handles.
     # Add them here and in to_disaggregated_params()/to_llm_disaggregated_params()
     # before routing MM encoder -> context -> generation through OpenAI protocol.
 
 
+class ConversationParams(OpenAIBaseModel):
+    model_config = ConfigDict(extra="forbid",
```
Comment (Shixiaowei02) [inline tensorrt_llm/serve/openai_protocol.py:167]: Removing `conversation_id` from the `extra="forbid"` serve `DisaggregatedParams` means clients (the historical scripts) still sending it would get an error.

File: tensorrt_llm/_torch/pyexecutor/_util.py (hunk 2 above)
Comment (jiaganc) [reply tensorrt_llm/_torch/pyexecutor/_util.py:486]: The conv aware adp router in the `feat/deepseek_v4` branch will read it, so we need it.

File: tensorrt_llm/serve/openai_protocol.py (hunk 3 above)
Comment (jiaganc) [reply tensorrt_llm/serve/openai_protocol.py:167]: The only client I know sending this is our internal benchmarking client, so I think it's safe to remove it?

**Comment by Shixiaowei02** [comment]
As noted above, this PR appears incomplete. I would suggest cherry-picking it back after merging to the feat branch. Thanks

**Comment by jiaganc** [comment]
This change will be merged to main along with the DSv4 merge back PR #15625. Close this PR.
